feat: drop support for php 8.0

st3iny opened the pull request:
## Summary

PHP 8.0 has been EOL for a long time now. Even debian stable is on PHP 8.2[^1].

This change should also be reflected in the documentation and workflow template repository.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

[^1]: https://packages.debian.org/bookworm/php

st3iny committed:
feat: drop support for php 8.0

Signed-off-by: Richard Steinmetz <richard@steinmetz.cloud>

nickvergessen reviewed (approved):
The missing platform was not Debian, but SLES.
After some more email communication they made it temporarily available on packagehub until 15 service pack 6 will ship it later this year:
https://packagehub.suse.com/packages/php81/

So this is okay now

come-nc commented on lib/versioncheck.php:
```diff
@@ -25,10 +25,10 @@
  */
-// Show warning if a PHP version below 8.0 is used,
-if (PHP_VERSION_ID < 80000) {
+// Show warning if a PHP version below 8.1 is used,
+if (PHP_VERSION_ID < 80100) {
```
`PhpOutdated` setup check also needs to be adapted or removed.
I would say we can make it warn when PHP is <8.2, since 8.1 will be EOL soon after next release?

st3iny replied:
I adjusted the check.

> You are currently running PHP 8.1.26. PHP 8.1 is now deprecated in Nextcloud 30. Nextcloud 31 may require at least PHP 8.2. Please upgrade to one of the officially supported PHP versions provided by the PHP Group as soon as possible. For more details see the [documentation ↗](https://secure.php.net/supported-versions.php).

st3iny committed:
ci(drone): use *-php8.3 images

Ref https://github.com/nextcloud/docker-ci/pull/634

Signed-off-by: Richard Steinmetz <richard@steinmetz.cloud>

come-nc commented:
Also drone images use 8.0, so we need to upgrade or migrate to github CI

come-nc committed:
fix: Fix new psalm errors from update

Not sure about the SimpleContainer modification, let’s see what CI says
 about that.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

st3iny commented:
Why does Psalm check files inside 3rdparty?

> 3rdparty/symfony/routing/Route.php:109:27: MethodSignatureMustProvideReturnType: Method Symfony\Component\Routing\Route::unserialize must have a return type signature!

come-nc commented:
> Why does Psalm check files inside 3rdparty?
> 
> > 3rdparty/symfony/routing/Route.php:109:27: MethodSignatureMustProvideReturnType: Method Symfony\Component\Routing\Route::unserialize must have a return type signature!

I think it always did, because we need to include 3rdparty files for class references, and we never added anything to ignore errors in there.
Add this one to the baseline.

come-nc committed:
chore: Add 3rdparty error to baseline. Also removed fixed errors.

The CDATA stuff comes from a psalm new version, nothing to worry about,
 both syntax are equivalent.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

come-nc commented:
Conflict because of https://github.com/nextcloud/server/pull/44663 fixing some of the same psalm errors xD

nickvergessen commented on lib/private/Authentication/TwoFactorAuth/Db/ProviderUserAssignmentDao.php:
```diff
@@ -91,32 +91,30 @@ public function persist(string $providerId, string $uid, int $enabled): void {
-				'enabled' => (int) $row['enabled'] === 1,
+				'provider_id' => (string)$row['provider_id'],
+				'uid' => (string)$row['uid'],
+				'enabled' => $row['enabled'] === 1,
```
```suggestion
				'enabled' => ((int) $row['enabled']) === 1,
```
? Some DBs only return strings

st3iny replied:
Ahh, that's why oci test are failing. Thanks!

AndyScherzinger committed:
chore(drone): Update drone file signature

Signed-off-by: Andy Scherzinger <info@andy-scherzinger.de>
Signed-off-by: Richard Steinmetz <richard@steinmetz.cloud>

skjnldsv commented:
docs in https://github.com/nextcloud/documentation/pull/11710
